src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -236,6 +236,29 @@ private MessagePayloadDataConverter CreateErrorDataConverter(IErrorSerializerSet
             return new MessagePayloadDataConverter(errorSerializerSettingsFactory.CreateJsonSerializerSettings(), isDefault);
         }
 
+        private IDurabilityProviderFactory GetDurabilityProviderFactory(ILogger logger, IEnumerable<IDurabilityProviderFactory> orchestrationServiceFactories)
+        {
+            bool storageTypeIsConfigured = this.Options.StorageProvider.TryGetValue("type", out object storageType);
+            if (!storageTypeIsConfigured)
+            {
+                logger.LogInformation("Using the default storage provider: Azure Storage.");
+                return orchestrationServiceFactories.FirstOrDefault(f => f.Name.Equals("Azure Storage"));
+            }
+
+            if (orchestrationServiceFactories.FirstOrDefault(f => f.Name.Equals(storageType)) == null)
+            {
+                IList<string> factoryNames = orchestrationServiceFactories.Select(f => f.Name).ToList();
+                logger.LogWarning($"Storage provider type ({storageType}) was not found. " +
+                    $"Available storage providers: {string.Join(", ", factoryNames)}. " +
+                    $"Using the default storage provider: Azure Storage.");
+
+                return orchestrationServiceFactories.FirstOrDefault(f => f.Name.Equals("Azure Storage"));

```

Comment:
I would probably use a global definition here, instead of a string literal, e.g. 
```csharp
const string DefaultProvider = "AzureStorage"
```